kdc/httpkadmind.c

```diff
@@ -1000,6 +1091,10 @@ get_keys1(kadmin_request_desc r, const char *pname)
     princ.tl_data = NULL;
 
     ret = krb5_parse_name(r->context, pname, &p);
+    if (ret == 0 && r->realm)
```

Comment:
Likely best to do something sensible when the parsed name already has a realm, and it isn't the one we use by default.
Perhaps use from SPN, or reject request, ... but avoid surprise...